plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/utils/RuntimeUtils.java

```diff
@@ -418,12 +415,21 @@ public static boolean isOSArchAMD64() {
     }
 
     /**
-     * Determine OS Sonoma
-     * 
-     * @return - true if Os Sonoma, version 14.x
+     * Retrieves version of the operating system.
+     *
+     * <p><b>Note:</b> this method was only tested to be working on macOS. Please test on other platforms.
+     *
+     * @return version of the operating system
      */
-    public static boolean isMacOsSomona() {
-        return isMacOS() && System.getProperty("os.version").startsWith("14");
+    @NotNull
+    public static Version getOSVersion() {
+        return new Version(System.getProperty("os.version"));
```

Comment:
new Version may throw exception. This may block app start.

Reply:
e.g. is it possible that OS version is 11.alpha ?

Reply:
As far as I am concerned, it always contains a string in the form of `major.minor[.micro]`.

I added a disclaimer about its reliability on different OSes.

This is why I wanted to use a more trustworthy source of that information, such as SWT Cocoa's OS class but we can't have a bundle dependency on it.

Reply:
We cannot be 100% sure. Lets handle exception at least